ocaml/typing/cmt2annot.ml

```diff
@@ -48,7 +48,11 @@ let bind_cases l =
         let open Location in
         match c_guard with
         | None -> c_rhs.exp_loc
-        | Some g -> {c_rhs.exp_loc with loc_start=g.exp_loc.loc_start}
+        | Some g ->
+          let gexp = match g with
+            | Predicate pred -> pred
+            | Pattern (exp, _, _) -> exp in
+          {c_rhs.exp_loc with loc_start=gexp.exp_loc.loc_start}
```

Comment:
I'm suspicious of this change. See the below call to `bind_variables`: it seems that binding is important, and pattern guards can bind variables. Do you know why/if this change is ok?

Reply:
Yeah, I do think we need to bind variables in guard patterns.

Modified to do this. I think the scopes that we have are always correct too.

Reply:
Yeah, my interpretation of "scope" is the location describing the extent of the bound variables in the pattern. What you have looks right.